src/java.base/share/classes/sun/nio/ch/FileChannelImpl.java

```diff
@@ -1133,10 +1138,41 @@ public MappedByteBuffer map(MapMode mode, long position, long size) throws IOExc
         }
     }
 
-    public Unmapper mapInternal(MapMode mode, long position, long size) throws IOException {
-        boolean isSync = isSync(Objects.requireNonNull(mode, "Mode is null"));
+    private static final int MAP_MEM_SEG_DEFAULT_MODES = 0;
+    private static final int MAP_MEM_SEG_READ_ONLY = 1;
+
+    @Override
+    public MemorySegment map(MapMode mode, long offset, long size,
+                             MemorySession session)
+            throws IOException
+    {
+        Objects.requireNonNull(mode,"Mode is null");
+        Objects.requireNonNull(session, "Session is null");
+        MemorySessionImpl sessionImpl = MemorySessionImpl.toSessionImpl(session);
+        sessionImpl.checkValidStateSlow();
+        if (offset < 0) throw new IllegalArgumentException("Requested bytes offset must be >= 0.");
+        if (size < 0) throw new IllegalArgumentException("Requested bytes size must be >= 0.");
+
+        boolean isSync = isSync(mode);
         int prot = toProt(mode);
-        return mapInternal(mode, position, size, prot, isSync);
+        Unmapper unmapper = mapInternal(mode, offset, size, prot, isSync);
+        int modes = MAP_MEM_SEG_DEFAULT_MODES;
+        if (mode == MapMode.READ_ONLY) {
+            modes |= MAP_MEM_SEG_READ_ONLY;
+        }
+        if (unmapper != null) {
+            AbstractMemorySegmentImpl segment = new MappedMemorySegmentImpl(unmapper.address(), unmapper, size,
```

Comment:
When reviewing the method for MappedByteBuffer: I think to make this consistent the "old" method should also use `address()` which applies the pagePosition. Currently it is confusing:
- New code returning `MemorySegment` uses `unmapper.address()`
- Old code returning `MappedByteBuffer` uses `unmapper.address + unmapper.pagePosition` (fields)

Should I open an issue or a PR to fix this (because this is already merged)?

See the mailing list posts:
- https://mail.openjdk.java.net/pipermail/panama-dev/2022-May/016981.html
- https://mail.openjdk.java.net/pipermail/panama-dev/2022-May/016990.html

Reply:
Please file an RFE. I suspect that there will be more little improvements and consolidation like this we'll want to make to this code.

Reply:
> RFE = issue?

issue, with type RFE (request for enhancement)

Reply:
See: JDK-8286734